src/components/Security/Users/List.vue

```diff
@@ -0,0 +1,396 @@
+        return
+      }
+      this.selectedDocuments = []
+      this.selectedDocuments = this.documents.map(document => document.id)
+    },
+    toggleSelectDocuments(id) {
+      let index = this.selectedDocuments.indexOf(id)
+
+      if (index === -1) {
+        this.selectedDocuments.push(id)
+        return
+      }
+
+      this.selectedDocuments.splice(index, 1)
+    },
+    onFiltersUpdated(newFilters) {
+      try {
+        filterManager.save(
+          newFilters,
+          this.$router,
+          this.index,
+          this.collection
+        )
+      } catch (error) {
+        this.$log.error(error)
+        this.$bvToast.toast('The complete error has been printed to console', {
+          title: 'Ooops! Something went wrong while updating the filters',
+          variant: 'warning',
+          toaster: 'b-toaster-bottom-right',
+          appendToast: true,
+          dismissible: true,
+          noAutoHide: true
+        })
+      }
+    },
+    async fetchDocuments() {
+      this.fetchingUsers = true
+
+      this.$forceUpdate()
+
+      this.selectedDocuments = []
+
+      let pagination = {
+        from: this.paginationFrom,
+        size: this.paginationSize
+      }
+
+      let searchQuery = null
+      searchQuery = filterManager.toSearchQuery(this.currentFilter)
+      if (!searchQuery) {
+        searchQuery = {}
+      }
+
+      let sorting = ['_id'] // by default, sort on _id: prevent random order
+      if (this.currentFilter.sorting) {
+        sorting = filterManager.toSort(this.currentFilter)
+      }
+
+      // TODO: refactor how search is done
+      // Execute search with corresponding searchQuery
+      try {
+        const res = await this.performSearch(
+          this.collection,
+          this.index,
+          searchQuery,
+          pagination,
+          sorting
+        )
+        this.documents = res.documents
+        this.totalDocuments = res.total
+      } catch (error) {
+        this.$log.error(error)
+        this.$log.debug(error.stack)
+        this.$bvToast.toast('The complete error has been printed to console', {
+          title: 'Ooops! Something went wrong while fetching users.',
+          variant: 'warning',
+          toaster: 'b-toaster-bottom-right',
+          appendToast: true,
+          dismissible: true,
+          noAutoHide: true
+        })
+      }
+      this.fetchingUsers = false
+    },
+    editUser(id) {
+      this.$router.push({
+        name: 'SecurityUsersUpdate',
+        params: { id }
+      })
+    },
+
+    // DELETE
+    // =========================================================================
+    async onDeleteConfirmed() {
+      this.deleteModalIsLoading = true
+      try {
+        await performDeleteUsers(
+          this.index,
+          this.collection,
+          this.candidatesForDeletion
+        )
+        this.$bvModal.hide('modal-delete-users')
+        this.deleteModalIsLoading = false
+        this.fetchDocuments()
```

Comment:
This call is not awaited nor the promise rejection catched

Reply:
Yes, it's because `fetchDocuments` already wraps sensitive code in a `try...catch` block and displays a toaster in case of error

Reply:
Yes but since you are not awaiting this call, the `fetchDocuments` method will not throw on error but just return a rejected promise which end up unhandled

Consider this code
```
async function fetchDocuments(b) {
  if (b) {
    throw new Error()
  }
}

async function run () {
  try {
    fetchDocuments(true)
  }
  catch (error) {
    console.error(error)
  }
}

run()
```
This will not be catched by the try catch and you will end up with a `unhandledRejection` error